x-pack/plugins/security_solution/public/management/components/endpoint_action_failure_message/endpoint_action_failure_message.tsx

```
@@ -24,51 +37,97 @@ export const EndpointActionFailureMessage = memo<EndpointActionFailureMessagePro
+          const endpointAgentOutput = action.outputs?.[agent];
+          const agentState = action.agentState[agent];
+          const hasErrors = agentState && agentState.errors;
+          const hasOutputCode: boolean =
+            !!endpointAgentOutput &&
+            endpointAgentOutput.type === 'json' &&
+            !!endpointAgentOutput.content &&
+            !!endpointAgentOutput.content.code;
+
+          const agentErrorInfo: { name: string; errors: string[] } = { name: '', errors: [] };
 
           if (
-            endpointAgentOutput &&
-            endpointAgentOutput.type === 'json' &&
-            endpointAgentOutput.content.code &&
-            endpointActionResponseCodes[endpointAgentOutput.content.code]
+            hasOutputCode &&
+            !!endpointAgentOutput &&
+            !!endpointActionResponseCodes[endpointAgentOutput.content.code]
           ) {
-            errors.push(endpointActionResponseCodes[endpointAgentOutput.content.code]);
+            agentErrorInfo.errors.push(
+              endpointActionResponseCodes[endpointAgentOutput.content.code]
+            );
+          }
+
+          if (hasErrors) {
+            const errorMessages: string[] = [...new Set(agentState.errors)];
+            agentErrorInfo.errors.push(...errorMessages);
+          }
+
+          if (agentErrorInfo.errors.length && action.hosts[agent].name) {
+            agentErrorInfo.name = action.hosts[agent].name;
+          }
+
+          if (agentErrorInfo.errors.length) {
+            allAgentErrors.push(agentErrorInfo);
           }
         }
       }
 
-      if (!errors.length) {
-        if (action.errors) {
-          errors.push(...action.errors);
-        } else {
-          errors.push(
-            i18n.translate('xpack.securitySolution.endpointActionFailureMessage.unknownFailure', {
-              defaultMessage: 'Action failed',
-            })
-          );
-        }
+      if (!allAgentErrors.length) {
+        return null;
       }
```

I think we should return a default error here - I see you removed the prior implementation. Maybe add it back and change the message to `Unknown error`??

Reasoning: if we reached this point in the component (got passed line 38, then the action is at error, so we should display something in the UI or else the user will not really see anything (also - we devs would have a hard time debugging issues like this if it was to come in via an SDH).

---

Oh yeah I missed that. We agreed on the reasoning earlier. I'll add it back.

---

852ac69f2829f817b0de1c1e7ccbc56bf77c8a60